### ISSUE TITLE
🐛 `assertion 'left == right' failed: Rewrite didn't consume all tokens`

### ISSUE THREAD

**Issue by qarmin** [biomejs/biome#4592]
### Environment information

```block
CLI:
  Version:                      0.0.0 (self compiled 6b183872744780ac55f7e6f45ac14f60b2796d49)
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           unset
  JS_RUNTIME_NAME:              unset
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       unset

Workspace:
  Open Documents:               0
```

### What happened?

File content(at the bottom should be attached raw, not formatted file - github removes some non-printable characters, so copying from here may not work):
```
delete<v l={{}i }
```

command
```
timeout -v 150 biome lint TEST___FILE.js
```

cause this
```
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates/biome_js_parser/src/parser/rewrite_parser.rs:95:9
Thread Name: biome::worker_1
Message: assertion `left == right` failed: Rewrite didn't consume all tokens
  left: 17
 right: 18


##### Automatic Fuzzer note, output status "None", output signal "Some(6)"
```

[compressed.zip](https://github.com/user-attachments/files/17816255/compressed.zip)

### Expected result

No crash

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by arendjr** [comment]
This issue is easily reproduced by placing the file from the Zip archive in `crates/biome_js_parser/tests/js_test_suite/error/issue4592.tsx` and running `cargo test` inside the `biome_js_parser` crate.
